### PR TITLE
Upgrade Insights Page

### LINKED ISSUE #57: Upgrade insights with month-over-month and cash-flow comparison
## Why it matters
The current insights page looks polished, but it is still too shallow analytically. A final project benefits a lot from one clear, thoughtful financial-analysis view.
## Scope
Add previous-vs-current month comparison cards, category delta callouts, and a real cash-flow visualization. This should explicitly include a time-based income-vs-expenses comparison or a net cash-flow trend panel for recent months or selected periods.
## Acceptance Criteria
- Insights includes a time-based cash-flow visualization, not only static totals.
- Users can see income and expenses compared over time, or clearly see net cash-flow trend.
- Previous-month comparison is available for the selected period.
- The chart is readable in both themes and on smaller screens.
- The new metrics are easy to explain in the final presentation and align with the numbers used elsewhere.
## Story Points
3
## Dependencies
Depends on #41

### PR DESCRIPTION
**Pull request by adillpickles**
## Description

Adds a deeper Insights experience: time-based cash-flow (income vs expenses / net), previous-vs-current month context for the selected period, and supporting analytics APIs/styles so the page matches the rest of the app and stays readable in both themes and on small screens.

## Related User Story / Issue

Fixes #57

## Type of Change

- [x] New feature
- [ ] Bug fix
- [x] Backend / API change
- [x] UI change
- [ ] Database change
- [ ] Documentation
- [ ] Refactor

## How Has This Been Tested?

- [x] GitHub Actions / CI tests passed
- [ ] Manual testing performed
- [ ] Not tested locally

### Test Notes

Added/updated Jest and API tests around Insights; rely on CI for full verification.

## UI Changes (if applicable)

- [ ] No UI changes
- [x] UI updated (add screenshots if needed)

## Screenshots (if UI changed)
image
image

## Checklist

- [x] Linked to a user story or issue
- [x] Code builds / tests pass in CI
- [x] Ready for review

### CODE REVIEW

**Comment by adillpickles** [comment]
Force pushed a cleaned up branch. The previous history accidentally included unrelated category/transaction commits from a stale local main. those have been removed. This PR now contains only the Insights work for Issue #57.

File: nextjs/src/lib/insights.js [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+function buildDailySpendDetails(rows = []) {
+  const grouped = new Map()
+
+  rows.forEach((row, index) => {
+    const key = formatDayKey(row.occurred_on)
+    if (!key) return
+
+    const visual = getCategoryVisual(row.category_name || row.title || `Expense ${index + 1}`)
+    const nextEntry = {
+      id: row.id ?? `daily-expense-${index}`,
```
Comment (zakariac22) [inline nextjs/src/lib/insights.js]: If row.id is ever null, the fallback ID depends on a query which cannot guarantee stability in case the result order changes. Use a more reliable fallback or ensure the row.id can be present.

File: nextjs/__tests__/frontend/insights.unit.test.js
Comment (zakariac22) [inline nextjs/__tests__/frontend/insights.unit.test.js:1]: Ensure testing edge cases such as boundary values of progress_percentage exactly at 0 or 100. Also test cases where functions handle empty or partially missing data.

File: nextjs/__tests__/frontend/insights-view.integration.test.js
Comment (zakariac22) [inline nextjs/__tests__/frontend/insights-view.integration.test.js:1]: Ensure naming clarity. This seems to be an integration test for insights-view.js. This may be confused with lib/insights.js. Also, use React Testing Library to render the frontend and test what should be expected.

File: nextjs/src/lib/insights.js (hunk 1 above)
Comment (adillpickles) [reply nextjs/src/lib/insights.js]: The query still selects the real expense id when available, but I replaced the index based fallback with a stable fallback derived from the day, amount, title, and category, with a suffix only for true duplicate fallback rows. I also added unit coverage for that behavior.

File: nextjs/__tests__/frontend/insights.unit.test.js
Comment (adillpickles) [reply nextjs/__tests__/frontend/insights.unit.test.js:1]: Added focused unit coverage for 0% and 100% boundaries, empty inputs, partial/missing data, empty cash flow/daily spend cases, and stable fallback daily expense ids.

File: nextjs/__tests__/frontend/insights-view.integration.test.js
Comment (adillpickles) [reply nextjs/__tests__/frontend/insights-view.integration.test.js:1]: Renamed the frontend render test to `insights-view.integration.test.js` so it is clearly for `insights-view.js`, and kept it under `__tests__/frontend/`. It uses React Testing Library render/screen assertions for the visible Insights UI.

File: nextjs/src/components/insights-view.js
Comment (zakariac22) [inline nextjs/src/components/insights-view.js:1]: When using light mode, the dark mode remains for insights-view. After this fix, PR is ready for approval.

Comment (adillpickles) [reply nextjs/src/components/insights-view.js:1]: fixed

**Review by zakariac22 — APPROVED** [review]
